images/linux/scripts/installers/rust.sh

```diff
@@ -19,19 +19,10 @@ source $CARGO_HOME/env
 rustup component add rustfmt clippy
 cargo install --locked bindgen cbindgen cargo-audit cargo-outdated
 
-# Permissions
-chmod -R 777 $(dirname $RUSTUP_HOME)
-
 # Cleanup Cargo cache
 rm -rf ${CARGO_HOME}/registry/*
 
 # Update /etc/environemnt
-prependEtcEnvironmentPath "${CARGO_HOME}/bin"
-
-# Rust Symlinks are added to a default profile /etc/skel
-pushd /etc/skel
-ln -sf $RUSTUP_HOME .rustup
-ln -sf $CARGO_HOME .cargo
-popd
+prependEtcEnvironmentPath '$HOME/.cargo/bin'
```

Comment:
I don't see any benefits in this approach, because you can't avoid invoking post-deployments scripts due to unknown $HOME variable. In fact, .rustup&.cargo are quite huge folders to multiple them at least two or three times.

Reply:
@maxim-lobanov don't mind invoking the script replacing $HOME afaik 😃

Reply:
Yea, replacing $HOME in `/etc/environment` is the part that we can't get rid of on self-hosted / premium runners. And it is not something that we can rework somehow.
So we replicate this script for Premium Runners. It is not big deal and takes little time so we can live with it.
We definitely can't maintain any big scripts on this stage (like having separate script for every tool) but replicating $HOME is some kind of core here.

P.S. We don't run https://github.com/actions/virtual-environments/blob/main/images/linux/post-generation/environment-variables.sh exactly but do pretty similar logic and this PR should work for us.